components/browser/awesomebar/src/test/java/mozilla/components/browser/awesomebar/BrowserAwesomeBarTest.kt

```diff
@@ -133,6 +134,43 @@ class BrowserAwesomeBarTest {
         }
     }
 
+    @Test
+    fun `removeProvider removes the provider`() {
+        runBlocking(testMainScope.coroutineContext) {
+            val provider1 = mockProvider()
+            val provider2 = mockProvider()
+            val provider3 = mockProvider()
+
+            val awesomeBar = BrowserAwesomeBar(context)
+            awesomeBar.addProviders(provider1, provider2)
+            awesomeBar.removeProviders(provider2)
+            awesomeBar.addProviders(provider3)
+
+            awesomeBar.onInputStarted()
+
+            verify(provider1).onInputStarted()
+            verify(provider2, never()).onInputStarted()
+            verify(provider3).onInputStarted()
+        }
+    }
+
+    @Test
+    fun `removeAllProviders removes all providers`() {
+        runBlocking(testMainScope.coroutineContext) {
+            val provider1 = mockProvider()
+            val provider2 = mockProvider()
+
+            val awesomeBar = BrowserAwesomeBar(context)
+            awesomeBar.addProviders(provider1, provider2)
+            awesomeBar.removeAllProviders()
+
+            awesomeBar.onInputStarted()
+
+            verify(provider1, never()).onInputStarted()
+            verify(provider2, never()).onInputStarted()
+        }
+    }
+
```

Comment:
Those test cases are great. Maybe we could also test that removing a provider will call `removeSuggestions` on the adapter? In this file you'll find examples where we replace the adapter with a mock to verify calls on the adapter.

Reply:
Good ideas on these tests--I added them!